app/javascript/components/upload/DifferentialExpressionFileForm.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+
+import { FileTypeExtensions, matchingFormFiles, validateFile } from './upload-utils'
+import { TextFormField } from './form-components'
+import ExpandableFileForm from './ExpandableFileForm'
+import Select from '~/lib/InstrumentedSelect'
+import { clusterFileFilter } from './ClusteringStep'
+import { differentialExpressionFileFilter } from './DifferentialExpressionStep'
+import CreatableSelect from 'react-select/creatable'
+
+const allowedFileExts = FileTypeExtensions.plainText
+
+/** renders a form for editing/uploading a differential expression file */
+export default function DifferentialExpressionFileForm({
+  file,
+  allFiles,
+  updateFile,
+  saveFile,
+  deleteFile,
+  bucketName,
+  isInitiallyExpanded,
+  isAnnDataExperience,
+  annotationsAvailOnStudy,
+  menuOptions
+}) {
+  // TODO (SCP-5154) Add DE specific clientside validation
+  const validationMessages = validateFile({ file, allFiles, allowedFileExts })
+
+  const fragmentType = isAnnDataExperience ? 'cluster' : null
+
+  const clusterFiles = matchingFormFiles(allFiles, clusterFileFilter, isAnnDataExperience, fragmentType)
+  const clusterFileOptions = clusterFiles.map(cf => ({ label: cf.name, value: cf._id }))
+  const associatedCluster = clusterFileOptions?.find(
+    opt => opt.value === file.differential_expression_file_info.clustering_association?.id
+  )
+
+  const annotsAlreadyInUse = []
+  // retrieve the annotations that are already in use on a DE file
+  allFiles.filter(differentialExpressionFileFilter).filter(
+    diffExpFile => diffExpFile.differential_expression_file_info.annotation_association?.length > 0
+  ).forEach(file => {
+    annotsAlreadyInUse.push(file.differential_expression_file_info.annotation_association[0])
+  })
+
+  // filter down the annotations to only allow choosing an annotation that hasn't been chosen already
+  // each annotation is allowed to be associated with only one DE file
+  const annotationOptions = annotationsAvailOnStudy?.map(
+    cf => ({ label: cf.name, value: cf.name })
```

Comment:
Blocker: if we're only matching on the value of `name`, we run the risk of confusing cluster- and study-based annotations that have the same name.  We need to be matching on all three annotation attributes: `name`, `type`, and `scope`.  Since this is DE, the `type` will always be `group`, but we still need to match on that.  Since this is a select, we should use the URL request parameter format, like `cell_type--group--study`.

Reply:
oh thanks for raising this, please see the latest commit and let me know if that addresses your concern: [652a709](https://github.com/broadinstitute/single_cell_portal_core/pull/1784/commits/652a70983472904e43a6ae2dc845be7078624fdf)